Haskell/Cabal.sublime-syntax

```diff
@@ -0,0 +1,406 @@
+
+  dependency-keys:
+    - match: ^(\s*)({{dependency_tags}})(:)
+      captures:
+        2: meta.mapping.key.cabal entity.name.tag.cabal
+        3: meta.mapping.cabal punctuation.separator.mapping.key-value.cabal
+      push: dependency-values
+
+  dependency-values:
+    - meta_scope: meta.dependencies.cabal
+    - meta_content_scope: meta.mapping.value.cabal
+    - include: value-end
+    - include: sequence-separators
+    - include: operators
+    - include: literal-versions
+    - include: ident-other
+
+###[ MODULES ]#################################################################
+
+  module-keys:
+    - match: ^(\s*)({{module_tags}})(:)
+      captures:
+        2: meta.mapping.key.cabal entity.name.tag.cabal
+        3: meta.mapping.cabal punctuation.separator.mapping.key-value.cabal
+      push: module-values
+
+  module-values:
+    - meta_scope: meta.modules.cabal
+    - meta_content_scope: meta.mapping.value.cabal
+    - include: value-end
+    - include: sequence-separators
+    - include: ident-modules
+
+###[ OPTIONS ]#################################################################
+
+  option-keys:
+    - match: ^(\s*)({{option_tags}})(:)
+      captures:
+        2: meta.mapping.key.cabal entity.name.tag.cabal
+        3: meta.mapping.cabal punctuation.separator.mapping.key-value.cabal
+      push: option-values
+
+  option-values:
+    - meta_scope: meta.options.cabal
+    - meta_content_scope: meta.mapping.value.cabal
+    - include: value-end
+    - include: literal-strings
+    - include: literal-options
+    - include: literal-versions
+
+###[ PATHS ]###################################################################
+
+  path-keys:
+    - match: ^(\s*)({{path_tags}})(:)
+      captures:
+        2: meta.mapping.key.cabal entity.name.tag.cabal
+        3: meta.mapping.cabal punctuation.separator.mapping.key-value.cabal
+      push: path-values
+
+  path-values:
+    - meta_scope: meta.paths.cabal
+    - meta_content_scope: meta.mapping.value.cabal
+    - include: value-end
+    - include: sequence-separators
+    - include: literal-paths
+
+###[ URLS ]####################################################################
+
+  url-keys:
+    - match: ^(\s*)({{url_tags}})(:)
+      captures:
+        2: meta.mapping.key.cabal entity.name.tag.cabal
+        3: meta.mapping.cabal punctuation.separator.mapping.key-value.cabal
+      push: url-values
+
+  url-values:
+    - meta_scope: meta.urls.cabal
+    - meta_content_scope: meta.mapping.value.cabal
+    - include: value-end
+    - include: sequence-separators
+    - include: literal-urls
+
+###[ VERSIONS ]################################################################
+
+  version-keys:
+    - match: ^(\s*)({{version_tags}})(:)
+      captures:
+        2: meta.mapping.key.cabal entity.name.tag.cabal
+        3: meta.mapping.cabal punctuation.separator.mapping.key-value.cabal
+      push: version-values
+
+  version-values:
+    - meta_scope: meta.versions.cabal
```

Comment:
Would it make sense to use `meta.section.*.cabal` for these? I don't remember a particular case where this was applied in the past, but it makes sense in my mind.

Reply:
A common 2nd level scope would probably make sense, yes.

1. I am just not sure how good `meta.section` vs. `meta.mapping` fit together to denote the  same block of code/text .
2. With regards to the example below `entity.name.section` is used for `name` in `executable name` lines. IMHO, the whole block below that line is a section of mappings. So it feels a bit strange to denote the key-value pairs as `section` as well, maybe.

**Example**

```
executable name
   compiler: ghc 8.3
   version: 3.3
   
executable othername
  compiler: ghc 8.4
  version 2.2
```

Reply:
Good point. If we're already in the values part of a mapping, then maybe we want to use `meta.mapping.value.*.cabal`?